File: Stack.cpp
//
//  Stack.cpp
//  Stack
//
//  Created by Евгений Киримов on 24.08.2018.
//  Copyright © 2018 Евгений Киримов. All rights reserved.
//

#include "Stack.hpp"
#include <iostream>

void Stack::push(int x){
    this->vec->push_back(x);
}

int Stack::pop(){
    vector<int>::iterator iter = this->vec->end();
    int value = * (iter-1);
    vec->erase(iter-1);
    return value;
    
}

bool Stack::isEmpty(){return this->vec->empty();}

int Stack::top(){
    vector<int>::iterator iter = this->vec->end();
    return *(iter-1);
    
}

void Stack::display(){
    for(int i=0; i < this->vec->size(); i++)
        std::cout << this->vec->at(i) << endl;
}

File: Stack.java
public class Stack {

    private int[] arr;
    private int top;


    Stack(int size){
        arr = new int[size];
        top = -1;
    }

    public void push(int x){
        arr[++top] = x;
    }

    public int pop(){return arr[top--];}

    public boolean isEmpty(){return top==-1;}

    public int top(){return arr[top];}

    public void display(){
        for(int i=0; i < arr.length; i++)
            System.out.print(arr[i]+" ");
        System.out.println();
    }
}
